localstack-core/localstack/testing/pytest/fixtures.py

```diff
@@ -904,10 +904,10 @@ def opensearch_wait_for_cluster(aws_client):
     def _wait_for_cluster(domain_name: str):
         def finished_processing():
             status = aws_client.opensearch.describe_domain(DomainName=domain_name)["DomainStatus"]
-            return status["Processing"] is False
+            return status["Processing"] is False and "Endpoint" in status
 
         assert poll_condition(
-            finished_processing, timeout=5 * 60
+            finished_processing, timeout=25 * 60, interval=20
```

Comment:
I guess AWS takes a long time to provision the domain? 

A 20s wait will increase the run time of this test dramatically when running against LocalStack. You should use conditional timeouts and values based on `localstack.testing.aws.util.is_aws_cloud()`

Reply:
Yes, provisioning takes about 20 minutes. I'll add the conditional. I also did some testing and found that we can significantly reduce the interval duration.